test/objects.spec.js

```diff
@@ -143,6 +143,36 @@ describe('Object Functions', () => {
       { x: 1, y: 2 },
     ])
   })
+  it('flattenObjectBy', () => {
+    // Flatten objects but leave arrays untouched
+    let flatten = F.flattenObjectBy(x =>
+      F.ifElse(_.isArray, _.map(flatten), undefined)(x)
+    )
+    expect(
+      flatten({
+        a: {
+          b: {
+            c: 1,
+          },
+        },
+      })
+    ).to.deep.equal({
+      'a.b.c': 1,
+    })
+    expect(
+      flatten([
```

Comment:
I would not expect `flattenObject` to return an array 🤔 

Is your expected behavior that this traverses into nested arrays of objects as well?

Reply:
It's weird, but it might just be what the client wants. If the customizer function returns a number at the top level what should we do? I think the sensible thing to do is to just trust whatever the user does